sys/net/gnrc/network_layer/ipv6/gnrc_ipv6.c

```diff
@@ -605,94 +606,87 @@ static void _send(gnrc_pktsnip_t *pkt, bool prep_hdr)
     hdr = ipv6->data;
     payload = ipv6->next;
 
-    gnrc_netif_t *netif = (iface == KERNEL_PID_UNDEF) ?
-                            NULL :
-                            gnrc_netif_get_by_pid(iface);
     if (ipv6_addr_is_multicast(&hdr->dst)) {
         _send_multicast(netif, pkt, ipv6, payload, prep_hdr);
     }
-    else if ((ipv6_addr_is_loopback(&hdr->dst)) ||      /* dst is loopback address */
-             ((iface == KERNEL_PID_UNDEF) && /* or dst registered to any local interface */
-              ((netif = gnrc_netif_get_by_ipv6_addr(&hdr->dst)) != NULL)
-             ) || ((iface != KERNEL_PID_UNDEF) && /* or dst registered to given interface */
-              (gnrc_netif_ipv6_addr_idx(netif, &hdr->dst) > 0)
-             )) {
-        uint8_t *rcv_data;
-        gnrc_pktsnip_t *ptr = ipv6, *rcv_pkt;
+    else {
+        gnrc_netif_t *tmp_netif = gnrc_netif_get_by_ipv6_addr(&hdr->dst);
 
-        if (prep_hdr) {
-            if (_fill_ipv6_hdr(netif, ipv6, payload) < 0) {
-                /* error on filling up header */
-                gnrc_pktbuf_release(pkt);
-                return;
+        if (ipv6_addr_is_loopback(&hdr->dst) ||    /* dst is loopback address */
+             /* or dst registered to a local interface */
+             (tmp_netif != NULL)) {
```

Comment:
Minor nit: Uncrustify is complaining here (indented 1 space too much).

Reply:
You mean the extra space? Can do, but with the line below, I would prefer to rather move it in by three more spaces (to be in line with the 4 space tabs, but also to distinguish it from the line below).

Reply:
I think uncrustify is going to complain about anything other than exactly four spaces from the `if`. With that I think it is best to keep it as it is as I agree that removing a space now decreases readability.

Reply:
> I think uncrustify is going to complain about anything other than exactly four spaces from the if. With that I think it is best to keep it as it is as I agree that removing a space now decreases readability.

That's why I proposed to do it this way:

```C
        if (ipv6_addr_is_loopback(&hdr->dst) ||    /* dst is loopback address */
                /* or dst registered to a local interface */
                (tmp_netif != NULL)) {
```

Reply:
I was trying to decrease (or not increase) the number of style issues generated by uncrustify, but from what I can see, uncrustify requires exactly (and not a multiple of) four spaces indentation here. 

It is not as if I don't agree with your proposal to increase readability, but making uncrustify happy here seems a bit impossible.

Reply:
> but from what I can see, uncrustify requires exactly (and not a multiple of) four spaces indentation here.

This is why I don't understand why you would think that my proposal would make uncrustify less happy (also `uncrustify`, at least in the current configuration, shouldn't define the status quo).

Reply:
Adapted it now as proposed.

Reply:
> This is why I don't understand why you would think that my proposal would make uncrustify less happy

Because it is still generating changes for these lines. Anyway, as you said, uncrustify is currently not leading in this. I'm okay with the current formatting.